vignettes/release-announcement.Rmd

```diff
@@ -0,0 +1,131 @@
+You can learn more about our concerns with the existing OOP systems in <https://rconsortium.github.io/OOP-WG/articles/motivation.html>.
+
+To avoid the problem of designing "yet another OOP system" for R, the design and implementation of S7 has been a team effort.
+The has been working as working group of R Consortium and including representatives from R-Core, BioConductor, Posit, and the wider R community.
+
+The long-term goal of this project is to get S7 in to base R, but for now, you can install it from CRAN:
+
+```{r, eval = FALSE}
+install.packages("S7")
+```
+
+## Usage
+
+Lets dive into the basics of S7.
+To learn more, read the packages vignettes including a more in-depth coverage of the the basics in [`vignette("S7")`](https://rconsortium.github.io/OOP-WG/articles/S7.html), generics and methods in [`vignette("generics-methods")`](https://rconsortium.github.io/OOP-WG/articles/generics-methods.html), and classes and objects in [`vignette("classes-objects")`](https://rconsortium.github.io/OOP-WG/articles/classes-objects.html).
+
+```{r}
+library(S7)
+```
+
+### Classes and objects
+
+S7 classes have a formal definition, specified by `new_class()` which includes a list of properties and an optional validator.
+For example, the following code creates a `range` class that includes `start` and `end` properties, and a validator that ensures that `start` is always less that `end`.
+
+```{r}
+range <- new_class("range",
+  properties = list(
+    start = class_double,
+    end = class_double
+  ),
+  validator = function(self) {
+    if (length(self@start) != 1) {
+      "@start must be length 1"
+    } else if (length(self@end) != 1) {
+      "@end must be length 1"
+    } else if (self@end < self@start) {
+      "@end must be greater than or equal to @start"
+    }
+  }
+)
+```
+
+`new_class()` returns the class object, which is also the function (aka constructor) you use to create instances of the class:
+
+```{r}
+x <- range(start = 1, end = 10)
+x
+```
+
+### Properties
+
+The data possessed by an object is called its **properties**.
+Use `@` to get and set properties:
+
+```{r}
+x@start
+
+x@end <- 20
+x
+```
+
+Properties are automatically validated against the type declared in `new_class()` (`double` in this case), and with the class **validator**:
+
+```{r, error = TRUE}
+x@end <- "x"
+
+x@end <- -1
+```
+
+### Generics and methods
+
+Like S3 and S4, S7 uses **functional OOP** where methods belong to **generic** functions, and method calls look like regular function calls: `generic(object, arg2, arg3)`.
+A generic uses the types of its arguments to automatically pick an appropriate implementation called a **method**.
+
+You can create a new generic with `new_generic()`: the first argument is the generic name (used in error messages) and the second gives the arguments used for dispatch.
+The third, and optional argument, supplies the body of the generic.
```

Comment:
```suggestion
The third, and optional, argument supplies the body of the generic.
```

Reply:
I thought Germans always wanted more commas 😄. I think optional might be better in a parenthetical here.

Reply:
Oh even better I can just put optional _before_ third.